src/main/java/dk/kb/datahandler/kaltura/KalturaDeltaUploadJob.java

```diff
@@ -0,0 +1,304 @@
+                // Can not fetch more records. Stop delta upload
+                moreSolrRecords=false;
+                log.error("Could not fetch more solr records from mTime:" + mTimeFromCurrent);                
+                throw new InternalServiceException("Could not fetch more solr records from mTime:" + mTimeFromCurrent);
+            }              
+        }
+        return numberStreamsUploaded;
+
+    }
+
+    private static void updateKalturaIdForRecord(DsStorageClient storageClient, String fileId, String kalturaInternalId) {     
+           storageClient.updateKalturaIdForRecord(fileId, kalturaInternalId);
+    }
+
+    /**
+     * Make Solr call to fetch records without a stream registered in kaltura.
+     * 
+     * @param mTimeFrom Only extract records with mTime higher that this value
+     * @param batchSize solr batch size.
+     * @return
+     * @throws SolrServerException
+     * @throws IOException
+     */
+    public static SolrDocumentList fetchSolrRecords(long mTimeFrom, int batchSize) throws SolrServerException, IOException {
+        String solrUrl = "http://ds-devel01.kb.dk:10007/solr/ds";
+        String filterQuery = "access_malfunction:false AND production_code_allowed:true AND NOT kaltura_id:*";  // only valid streams that does not have kaltura id already
+        SolrClient client = new Http2SolrClient.Builder(solrUrl).withConnectionTimeout(1, TimeUnit.MINUTES).build();
+
+        String query = "internal_storage_mTime:[" + mTimeFrom + " TO *]"; // mTimeFrom must start with this value or higher.
+        String fieldList = "title,description,file_id,id,resource_description,originates_from,internal_storage_mTime"; // only extract fields we need
+        System.out.println(query);
+
+        try (client) { // autoclosable
+            SolrQuery solrQuery = new SolrQuery();
+            solrQuery.setQuery(query);
+            solrQuery.setFilterQueries(filterQuery);
+            solrQuery.set("facet", "false"); // very important. Must overwrite to false. Facets are very slow and expensive.
+            solrQuery.set("hl",false);// no highlights
+            solrQuery.set("spellcheck", false); //No spellcheck
+            solrQuery.add("sort", "internal_storage_mTime ASC"); // increasing order
+            solrQuery.add("fl", fieldList);
+            solrQuery.setRows(batchSize);
+            QueryResponse response = client.query(solrQuery);
+            SolrDocumentList results = response.getResults();
+            log.info("Load solr records for delta upload:" + results.getNumFound());
+            return results;
+        }
+    }
+
+    /**
+     * 
+     * 
+     * @param title  Title field for kaltura
+     * @param description Description field for kaltura
+     * @param fileId  This value will be used to calculate path of stream depending on originatesFrom
+     * @param originatesFrom      DOMS or Preservica
+     * @param resourceDescription VideoObject or AudioObjext
```

Comment:
Please update names of these params in the javadoc to match the params given to the method.

Reply:
Fixed